cpp/include/cudf/detail/copy.hpp

```diff
@@ -115,6 +115,49 @@ std::unique_ptr<column> copy_if_else( column_view const& lhs, column_view const&
                                     rmm::mr::device_memory_resource *mr = rmm::mr::get_default_resource(),
                                     cudaStream_t stream = 0);
 
+/**
+ * @brief   Returns a new column, where each element is selected from either @p lhs or 
+ *          @p rhs based on the value of the corresponding element in @p boolean_mask
+ *
+ * Selects each element i in the output column from either @p rhs or @p lhs using the following rule:
+ *          output[i] = (boolean_mask[i]) ? lhs : rhs[i]
+ *         
+ * @throws cudf::logic_error if lhs and rhs are not of the same type 
+ * @throws cudf::logic_error if boolean mask is not of type bool8
+ * @throws cudf::logic_error if boolean mask is not of the same length as rhs  
+ * @param[in] left-hand scalar
+ * @param[in] right-hand column_view
+ * @param[in] column_view representing "left (true) / right (false)" boolean for each element
+ * @param[in] invert_mask whether the values of the boolean mask should be inverted (to allow for swapping order of lhs and rhs)
+ * @param[in] mr resource for allocating device memory
+ * @param[in] stream Optional CUDA stream on which to execute kernels
+ *
+ * @returns new column with the selected elements
+ */
+std::unique_ptr<column> copy_if_else( scalar const& lhs, column_view const& rhs, column_view const& boolean_mask, bool invert_mask,
```

Comment:
I didn't find the definition for the detail::copy_if_else, can you please point it out.

Reply:
It's in  cudf/detail/copy_if_else.cuh

Reply:
I could only find this https://github.com/rapidsai/cudf/blob/2014c9d45d1800ad3549bd2c33c7fb9670fb9dfb/cpp/include/cudf/detail/copy_if_else.cuh#L118

Reply:
Yes, that's the right one.

Reply:
But arguments are different

Reply:
Right.  The detail implementation exposes an iterator-only interface so that you can call into it with different combinations of types (columns, scalars).

Reply:
Updated.